### ISSUE TITLE
[package] glib: Can't have multiple dependencies linking against static glib in Windows

### ISSUE THREAD

**Issue by hesham-essam** [conan-io/conan-center-index#11022]
For all intents and purposes, static libraries with global state are generally broken in Windows.
If we have two shared libraries linking against a static library, then each one of these two libraries will get its own version of the global state. This is unlike on Linux.

The problem here is that `glib` relies heavily on global state, and many of the dependencies of gtk link against glib. If any of these dependencies is a shared library (which is actually the case with cairo and gdk-pixbuf), it will get its own version of glib separate from the others.
https://github.com/conan-io/conan-center-index/blob/e94a759bee152f96dcc2d6b67029bb137a043997/recipes/gtk/all/conanfile.py#L64-L69

I suggest that we make glib (and any other library with global state for that matter) shared by default on Windows.

Sorry I'm not sure if this is the right repository for this issue.

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **glib/all**
  * Operating System+version: **Windows*
  * Compiler+version: **msvc**

**Comment by SSE4** [comment]
that's true, in general you cannot have multiple instances of GLib in your address space, and it bites not only on Windows.
we faced exactly this problem with GStremer already, where in general static/shared GLib/GStreamer cannot be mixed.
we marked these combinations as invalid configurations to prevent such bugs.
see also: #7142 #7734 #8148 #10539 and https://github.com/conan-io/conan/issues/9712

**Comment by hesham-essam** [comment]
Do you think making glib shared by default a good solution? Glib works correctly as shared in most configurations, while it fails if static on many configurations.

**Comment by SSE4** [comment]
> Do you think making glib shared by default a good solution? Glib works correctly as shared in most configurations, while it fails if static on many configurations.

if it's what upstream recommends, then yes.

**Comment by AndreyMlashkin** [comment]
@memsharded  

Can conan client recognize such static+shared dependencies problems? Maybe it would be a good feature to implement a warning? Version override is already logged as a warning, it could be the same.

**Comment by uilianries** [comment]
The PR https://github.com/conan-io/conan-center-index/pull/29382 intragated GTK-4, limitating only shared library + shared dependencies to avoid missing static + shared and resulting in linkage errors when consuming GTK.
